src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +173,113 @@ export const handleRedirect: RequestHandler<
     redirectPath,
   })
 }
+
+/**
+ * Handler for GET /:formId/publicform endpoint
+ * @returns 200 if the form exists
+ * @returns 404 if form with formId does not exist or is private
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs or if the type of error is unknown
+ */
+export const handleGetPublicForm: RequestHandler<{ formId: string }> = async (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+
+  const formResult = await getFormIfPublic(formId).andThen((form) =>
+    FormService.checkFormSubmissionLimitAndDeactivateForm(form),
+  )
+
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
+    const { error } = formResult
+    logger.error({
+      message: 'Error retrieving public form',
+      meta: {
+        action: 'handleGetPublicForm',
+        ...createReqMeta(req),
+        formId,
+      },
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicForm = form.getPublicView()
+  const { authType } = form
+  let myInfoError = false
+
+  // NOTE: Creating a variable to ensure that TS will enforce the type and ensure all keys in AuthType are covered.
+  const formController: FormController<
```

Comment:
this should be defined as a helper function, rather than a variable which is re-initialised every time this controller function is run

Reply:
Shouldn't be a big deal, we create objects all the time; all  this is doing is creating an object to a bunch of references, not actually running the functions. Separating it out means we will not be able to act on `res.cookie` in the controller and adds complexity?

Reply:
> we create objects all the time

you're being reductive here - can you show me any other places in the backend where we use this particular pattern of inlining an object where the keys map to neverthrow chains? anyway sure I'll defer to your judgement here